File: ruby/reverse_whois_classic.rb
require 'erb'
require 'json'
require 'net/https'
require 'uri'
require 'yaml' # only needed to print the returned result in a very pretty way

########################
# Fill in your details #
########################
username = 'Your reverse whois api username'
password = 'Your reverse whois api password'
term = 'wikimedia'

url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php' \
      '?term1=' + ERB::Util.url_encode(term) +
      '&username=' + ERB::Util.url_encode(username) +
      '&password=' + ERB::Util.url_encode(password) +
      '&output_format='

#######################
# Use a JSON resource #
#######################
format = 'json'

# Open the resource
buffer = Net::HTTP.get(URI.parse(url + format))

# Parse the JSON result
result = JSON.parse(buffer)

# Print out a nice informative string
puts "JSON:\n" + result.to_yaml + "\n"

File: node/reverse_whois_classic.js
var http = require('https');
var querystring = require('querystring');

var term = 'cinema';
var username = 'your reverse whois api username';
var password = 'your reverse whois api password';
var mode = 'preview';
var rows = 20;

var url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php';

var params = {
    term1: term,
    username: username,
    password: password,
    mode: mode,
    rows: rows
};

url += '?' + querystring.stringify(params);

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) {
        str += chunk;
    });
    response.on('end', function() {
        console.log(JSON.parse(str));
    });
}).end();